### PR TITLE
Feat: Multi modal example

### PR DESCRIPTION
**Pull request by BastinJafari**
<h2>PR checklist</h2>

**Docs**
- [ ] If you added new code; is it self documenting? (all the variable names clear, comments added where appropriate)
- [ ] If you touched existing code; did you refactor existing naming, added comments where necessary?

**Tests**
- [ ] If your ticket adds new functionality; have you added unit tests to verify behaviour?
- [ ] If your ticket alters user interaction with the UI; have you added integration tests?
- [ ] Do your tests check for every _convievable_ user behabviour? (all behaviours the user _can_ perform)

**Typing**
- [ ] If you added new code; have you added typing?
- [ ] If you touched existing code; have you improved its typing?

**Styling**
- [ ] If new styles were created; Did you refactor/add more styles in css-in-js style?

### CODE REVIEW

File: src/redux/flows/flows.actions.ts [hunk 1]
```diff
@@ -198,91 +193,114 @@ export function updateSelectedWorkspace(
...
-      flow.workspaceId,
-      "\n\tflowId: ",
-      flowId,
-      "\n\tchart:",
-      flowChart,
-      "\n\tyaml:",
-      yaml
-    )
+    if (!workspaceDaemonId) {
+      const creationResult = await jinadClient.createWorkspace(["empty_blob"]) //Creating a workspace in daemon requires files
```
Comment (aga11313) [inline src/redux/flows/flows.actions.ts:201]: Does it just pass an empty file here? What happens to this file afterwards

File: src/services/multiModalScript.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  logger.log(flowResult, "flowResult")
+
+  const settings = store.getState().settingsState.settings
+  const gatewayResult = await gatewayClient.connect(settings, (result) =>
+    logger.log(result)
+  )
+
+  logger.log(gatewayResult, "gatewayResult")
+
+  for (let i = 0; i < 10; i++) {
```
Comment (aga11313) [inline src/services/multiModalScript.ts:37]: is this some remnant of testing?

File: src/services/multiModalScript.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      id: indexData[i]["1"].toString(),
+      image: indexData[i]["image_1.jpg"],
+      caption:
+        indexData[i]["A beautiful young girl posing on a white background."],
+    }
+
+    await gatewayClient.index(JSON.stringify(data))
+  }
+
+  console.log("terminateFlow")
```
Comment (aga11313) [inline src/services/multiModalScript.ts:49]: A bunch of console.log we should probably get rid of

File: src/services/tests/gatewayClient.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+    if (gatewayInstance) {
+      gatewayInstance.defaults.baseURL = baseURL
+      gatewayInstance.defaults.timeout = TIMEOUT
+    } else gatewayInstance = axios.create({ baseURL, timeout: TIMEOUT })
+
+    try {
+      const result = await gatewayInstance.get("/status")
+      if (result.status === 200) {
+        logger.log("api - connect successfully connected to gateway")
```
Comment (aga11313) [inline src/services/tests/gatewayClient.ts:22]:
```suggestion
        logger.log("api - successfully connected to gateway")
```

**Review by aga11313 — COMMENTED** [review]
lgtm but waiting on @Roshanjossey to review

**Review by Roshanjossey — APPROVED** [review]

File: src/data/multiModalExample.ts [hunk 5]
```diff
@@ -0,0 +1,18538 @@
+export const configData = {
+  "__init__.py": "",
+
+  "crafter-image.yml":
```
[the remark this answers is not in the document]
Comment (BastinJafari) [reply src/data/multiModalExample.ts]: Not anymore!

**Review by aga11313 — APPROVED** [review]

**Pull request closed without merging**